### PR TITLE
Resnet only use contiguous in training mode.

### LINKED ISSUE #12975: AnimateDiffPipeline performance regression on CPU.
### Describe the bug

regression PR: https://github.com/huggingface/diffusers/pull/11098 .

The [attn_processor](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_processor.py#L2778) will turn the hidden_states into channel last layout, which will have great benefit on CPU when running matmul.

But the op: `.contiguous()` will turn the tensor into NCHW layout, it will make the matmul slower than the original NHWC layout.

### Reproduction

numactl -C 0-31 --membind 0 python test.py
```python
from diffusers import AnimateDiffPipeline, MotionAdapter, EulerDiscreteScheduler
from safetensors.torch import load_file
from huggingface_hub import hf_hub_download
from transformers import set_seed
import torch
import time

SEED = 42
device = "cpu"
model_dtype = torch.float16
WARM_UP = 4
RUN = 4

set_seed(SEED)

print("\nLoading AnimateDiff-Lightning model...")
model_id = "ByteDance/AnimateDiff-Lightning"
step = 4
ckpt = f"animatediff_lightning_{step}step_diffusers.safetensors"
base = "emilianJR/epiCRealism"

adapter = MotionAdapter().to(device, model_dtype)
adapter.load_state_dict(load_file(hf_hub_download(model_id, ckpt), device=device))
pipe = AnimateDiffPipeline.from_pretrained(base, motion_adapter=adapter, torch_dtype=model_dtype).to(device)
pipe.scheduler = EulerDiscreteScheduler.from_config(
    pipe.scheduler.config, timestep_spacing="trailing", beta_schedule="linear"
)

def run_inference():
    set_seed(SEED)
    with torch.no_grad():
        output = pipe(
            prompt="An astronaut riding a green horse",
            guidance_scale=1.0,
            num_inference_steps=4,
        ).frames[0]
    return output

# Warm up
print(f"\nWarming up ({WARM_UP} iterations)...")
for i in range(WARM_UP):
    run_inference()
    print(f"  Warm-up {i+1}/{WARM_UP} done")

# Benchmark
print(f"\nRunning benchmark ({RUN} iterations)...")
elapsed_times = []
for i in range(RUN):
    start = time.perf_counter()
    output = run_inference()
    end = time.perf_counter()
    elapsed = (end - start) * 1000  # ms
    elapsed_times.append(elapsed)
    print(f"  Run {i+1}/{RUN}: {elapsed:.2f} ms")

# Statistics
avg_time = sum(elapsed_times) / len(elapsed_times)
min_time = min(elapsed_times)
max_time = max(elapsed_times)
print(f"\n{'='*50}")
print(f"Results ({RUN} runs):")
print(f"  Average: {avg_time:.2f} ms")
print(f"  Min:     {min_time:.2f} ms")
print(f"  Max:     {max_time:.2f} ms")
print(f"{'='*50}")
```

The pipeline latency has 50% performance regression after the regression PR.

Since the PR is targeted to fix the DDP issue. I think we can check `if DDP` before using `.contiguous()`. WDYT? @sayakpaul 

Hi @jinc7461 . Could you please provide the script to reproduce the error, and give me some advice to check before using `.contiguous()` ? Thanks!

cc @sywangyi 

### Logs

```shell

```

### System Info

torch                     2.11.0.dev20260113+cpu
platform: Intel Xeon 6

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by jiqing-feng**
As the contiguous may have a negative impact on CPU performance, ResNet can only use it in training mode.
Fix #12975

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Thanks!

**Comment by sayakpaul** [comment]
The [failing test](https://github.com/huggingface/diffusers/actions/runs/21017138613/job/60430765224?pr=12977#step:6:1055) passes on `main`.

But with this PR, that fails. Could you please check?

**Comment by jiqing-feng** [comment]
The behaviour change has little impact on some models' precision; just changing the tolerance from `1e-4` to `2e-4` will be fine.
Hi @sayakpaul . Please re-run the CI. Thanks!

**Comment by sayakpaul** [comment]
But we need the tests to pass on our CI.

**Comment by jiqing-feng** [comment]
I will check these failed tests.

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . The failed test seems network issue. Please check it. Thanks!

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . Please retrigger the CI. Thanks!

**Comment by jiqing-feng** [comment]
It's so weird. The CI already passed in commit [aa65c5c](https://github.com/huggingface/diffusers/pull/12977/commits/aa65c5c1f5323b27707eaece3f97732c31327212), but failed after update branch.
image

@sayakpaul . Do you have any clue about it?

**Comment by jiqing-feng** [comment]
> It's so weird. The CI already passed in commit [aa65c5c](https://github.com/huggingface/diffusers/pull/12977/commits/aa65c5c1f5323b27707eaece3f97732c31327212), but failed after update branch. image
> 
> @sayakpaul . Do you have any clue about it?

Hi @sayakpaul . Would you please re-run the CI to see if the error disappears? Thanks!

**Comment by jiqing-feng** [comment]
I see the lora tests also failed in https://github.com/huggingface/diffusers/pull/13003

File: tests/pipelines/test_pipelines_common.py [hunk 1]
```diff
@@ -2212,7 +2212,7 @@ def test_StableDiffusionMixin_component(self):
 
     @require_hf_hub_version_greater("0.26.5")
     @require_transformers_version_greater("4.47.1")
-    def test_save_load_dduf(self, atol=1e-4, rtol=1e-4):
+    def test_save_load_dduf(self, atol=1e-3, rtol=1e-3):
```
Comment (sayakpaul) [inline tests/pipelines/test_pipelines_common.py]: We shouldn't change it at the main test level. If a model test failing, we should override the method at the corresponding test class and relax tolerance.

Comment (jiqing-feng) [reply tests/pipelines/test_pipelines_common.py]: Hi @sayakpaul . It was my mistake. I've already fixed it. Please review the new change. Thanks!

**Review by sayakpaul — APPROVED** [review]
Thanks!

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . Would you please rerun the CI? I can pass the failed CI locally and the failed CI didn't show before. BTW, I saw many CI failed in the last commit in the main branch. Please let me know what need to be changed before merge.

**Comment by sayakpaul** [comment]
Can you push an empty commit?

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . I have updated the branch. Please rerun the CI. Thanks!

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . I relax the tol for `tests/pipelines/kandinsky3/test_kandinsky3_img2img.py::Kandinsky3Img2ImgPipelineFastTests::test_save_load_dduf`. Could you please rerun the CI? Thanks!

**Comment by jiqing-feng** [comment]
Hi @sayakpaul. Is it okay to merge the PR? The new failed CI seems an accident because the previous commit can pass:
image
